Sources/BraintreeSEPADirectDebit/BTSEPADirectDebitRequest.swift

```diff
@@ -16,7 +37,7 @@ import BraintreeCore
     /// Required. The customer ID.
     public var customerID: String?
     
-    /// Required. The `BTSEPADebitMandateType`.
+    /// Optional. The `BTSEPADebitMandateType`. If not set, defaults to `.oneOff`
     public var mandateType: BTSEPADirectDebitMandateType?
```

Comment:
nit: I noticed on Android it's non-null with `ONE_OFF` as the default value. Would it make sense to do the same here and avoid the optional type?

Reply:
Yeah, so this is a bit weird. With all of the requests on the SDK, we currently default all of the values to `nil`  and make them optional so that we can do this syntax:
```
let sepaDirectDebitRequest = BTSEPADirectDebitRequest()
sepaDirectDebitRequest.accountHolderName = "John Doe"
sepaDirectDebitRequest. = ...
```
If we don't make them optional, they would be required when you create the object like:
```
let sepaDirectDebitRequest = BTSEPADirectDebitRequest(
    accountHolderName: "John Doe",
    ...
)
```
The latter is the more "Swift-y" way to do it, but it feels odd to make this change for 1 request type in Swift when the others can all use the `.` syntax. It also felt weird to make 1 part of the request non-optional and required when creating the Swift request when the rest of it could use the `.` syntax. We should explore which pattern we want to move forward with as we convert other modules.

We are still defaulting to `.onOff` in the init if not provided by the merchant.

Reply:
```suggestion
    public var mandateType: BTSEPADirectDebitMandateType = .oneOff
```
Yeah we did the same thing in Android [SEPADirectDebitRequest.java](https://github.com/braintree/braintree_android/blob/40e579ec02ede738aeab774ce31e187335f2d104/SEPADirectDebit/src/main/java/com/braintreepayments/api/SEPADirectDebitRequest.java#L14) we could do it like this in Swift.

Reply:
Yeah, so in Swift those defaults are provided in the init instead, [which we currently do in this PR](https://github.com/braintree/braintree_ios/blob/4a6cf68f9bb398a9d05784c72cf5fa2fd58c44cf/Sources/BraintreeSEPADirectDebit/BTSEPADirectDebitRequest.swift#L66). If we moved to this, even if we set a default, the merchant would have to set the mandate like so even if using the `.` syntax:
```
let sepaDirectDebitRequest = BTSEPADirectDebitRequest(mandateType: .oneOff)
sepaDirectDebitRequest.accountHolderName = "John Doe"
sepaDirectDebitRequest. = ...
```

Reply:
```suggestion
    public var mandateType: BTSEPADirectDebitMandateType
```
Oh word yeah my nit was actually with the type being optional. Having it in `init` is smooth, can we do this 👆 instead?

Reply:
I see cases for both here though but I've always felt like the less optionals the better. On the other hand, it does align nicely with the fact that the merchant isn't required to set this property.

Reply:
Ah gotcha gotcha. Yeah, I agree. I could go either way with this, honestly. I lean slightly more to defaulting all to optional now to match what we do in obj-c, but would love to move away from optionals as we decide on a pattern moving forward with Swift conversions. Would be interesting to start up a GH discussion to see how many merchants are actually using the `.` syntax since I feel like that could influence our decisions moving forward.